### ISSUE TITLE
Nested Rollable Tables do not work

### ISSUE THREAD

**Issue by Centor1982** [jopeek/fvtt-loot-sheet-npc-5e#293]
**Describe the bug**
I have created a rollable table that calls two rollable tables that are provided in the compendium to test. Specifically the Arcane Shop and Potion Shop. The table is called Magic Item Merchant and should roll the table then roll the nested tables (Arcane Shop, Potion Shop) to get the items and add them to the merchant's inventory.

**To Reproduce**
1 Create a Rollable table with these settings.:
![image](https://user-images.githubusercontent.com/88258067/154354756-64f4e7e8-7d48-492f-b366-deccf379599b.png)
2. Click on Rolltabel to test to chat (it works for me)
3. Setup a merchant with the table as the one created
4. Run update for Inventory

**Expected behavior**
The table would roll as normal and provide results from nested tables as it would to chat.

**Screenshots**
`lootsheetnpc5e |  \-->  | Filtered for players with a token in the tokens scene. 0 players with eligable characters.
LootSeeder.js:108 {itemQtyFormula: '2', itemQtyLimitFormula: '1d2', shopQtyFormula: '5', currencyFormula: '1d8[pp],(20d+2)*10 [gp], (10d+1)*10[sp]'}
CurrencyHelper.js:112 lootsheetnpc5e | currencyHelper : Error: Unresolved StringTerm 20d requested for evaluation
    at StringTerm.evaluate (foundry.js:9180:11)
    at Roll._evaluate (foundry.js:6795:42)
    at Roll.evaluate (foundry.js:6762:25)
    at ParentheticalTerm._evaluate (foundry.js:8798:28)
    at ParentheticalTerm.evaluate (foundry.js:7693:25)
    at Roll._evaluate (foundry.js:6782:20)
    at Roll.evaluate (foundry.js:6762:25)
    at Roll.roll (foundry.js:6862:17)
    at Function.tryRoll (CurrencyHelper.js:110:51)
    at Function.generateCurrency (CurrencyHelper.js:64:53)
    at async LootProcessor.buildResults (LootProcessor.js:41:14)
    at async Function.addLootToTarget (ActorHelper.js:171:17)
    at async Function.seedItemsToActors (LootSeeder.js:35:109)
    at async SheetListener.inventoryUpdateListener (SheetListener.js:164:62)
tryRoll @ CurrencyHelper.js:112
await in tryRoll (async)
generateCurrency @ CurrencyHelper.js:64
await in generateCurrency (async)
buildResults @ LootProcessor.js:41
addLootToTarget @ ActorHelper.js:171
await in addLootToTarget (async)
seedItemsToActors @ LootSeeder.js:36
await in seedItemsToActors (async)
inventoryUpdateListener @ SheetListener.js:167
await in inventoryUpdateListener (async)
(anonymous) @ SheetListener.js:90
CurrencyHelper.js:112 lootsheetnpc5e | currencyHelper : Error: Unresolved StringTerm 10d requested for evaluation
    at StringTerm.evaluate (foundry.js:9180:11)
    at Roll._evaluate (foundry.js:6795:42)
    at Roll.evaluate (foundry.js:6762:25)
    at ParentheticalTerm._evaluate (foundry.js:8798:28)
    at ParentheticalTerm.evaluate (foundry.js:7693:25)
    at Roll._evaluate (foundry.js:6782:20)
    at Roll.evaluate (foundry.js:6762:25)
    at Roll.roll (foundry.js:6862:17)
    at Function.tryRoll (CurrencyHelper.js:110:51)
    at Function.generateCurrency (CurrencyHelper.js:64:53)
    at async LootProcessor.buildResults (LootProcessor.js:41:14)
    at async Function.addLootToTarget (ActorHelper.js:171:17)
    at async Function.seedItemsToActors (LootSeeder.js:35:109)
    at async SheetListener.inventoryUpdateListener (SheetListener.js:164:62)
tryRoll @ CurrencyHelper.js:112
await in tryRoll (async)
generateCurrency @ CurrencyHelper.js:64
await in generateCurrency (async)
buildResults @ LootProcessor.js:41
addLootToTarget @ ActorHelper.js:171
await in addLootToTarget (async)
seedItemsToActors @ LootSeeder.js:36
await in seedItemsToActors (async)
inventoryUpdateListener @ SheetListener.js:167
await in inventoryUpdateListener (async)
(anonymous) @ SheetListener.js:90
SocketListener.js:42 lootsheetnpc5e | {this.name} | data {action: 'sheetUpdate', triggerActorId: null, tokenUuid: 'Actor.em8DhkwAwnpaS3tu', processorId: 'LnfrgJkRfBMtTMBR', targetItemId: null, …}
PermissionHelper.js:157 lootsheetnpc5e | PermissionsHelper | getEligablePlayerActors | 8 players with eligable characters found.
PermissionHelper.js:165 lootsheetnpc5e |  \-->  | Filtered for players with a token in the tokens scene. 0 players with eligable characters.
LootProcessor.js:293 Uncaught (in promise) TypeError: Cannot set properties of undefined (setting 'quantity')
[Detected 1 package: lootsheetnpc5e]
    at LootProcessor._addLootItem (LootProcessor.js:293:31)
    at async LootProcessor.addItemsToActor (LootProcessor.js:397:33)
_addLootItem @ LootProcessor.js:293
await in _addLootItem (async)
addLootToTarget @ ActorHelper.js:174
await in addLootToTarget (async)
seedItemsToActors @ LootSeeder.js:36
await in seedItemsToActors (async)
inventoryUpdateListener @ SheetListener.js:167
await in inventoryUpdateListener (async)
(anonymous) @ SheetListener.js:90
foundry.js:11928 Foundry VTT | Constructed index of foundry_community_tables.community-tables-5e Compendium containing 1 entries
foundry.js:11928 Foundry VTT | Constructed index of foundry_community_tables.community-tables-independent Compendium containing 184 entries
foundry.js:11928 Foundry VTT | Constructed index of lootsheetnpc5e.lootsheetnpc5e-rolltables Compendium containing 13 entries
foundry.js:11928 Foundry VTT | Constructed index of my-shared-content.shared-roll-tables Compendium containing 1188 entries
foundry.js:11928 Foundry VTT | Constructed index of token-mold.adjectives Compendium containing 7 entries
PermissionHelper.js:157 lootsheetnpc5e | PermissionsHelper | getEligablePlayerActors | 8 players with eligable characters found.
PermissionHelper.js:165 lootsheetnpc5e |  \-->  | Filtered for players with a token in the tokens scene. 0 players with eligable characters.`

**Desktop (please complete the following information):**
 - OS: Win10
 - Browser Chrome
 - Version 3.5.7 Foundry 9.249

**Additional context**
I have additional tables I would like to use to pull in magic items from a set of tables like the ones in the DMG but I paired it back to just these that I knew worked alone. And still got the error.

**Comment by DanielBoettner** [comment]
Please test if this is fixed in https://github.com/jopeek/fvtt-loot-sheet-npc-5e/releases/tag/v3.5.8

**Comment by Centor1982** [comment]
OK, I updated to 3.5.9 which I hope doesn't invalidate the test. I attempted to just change the table to the one I was using and it still didn't work but did give an error on screen that stated "There are no available results that can be drawn from this table". So I verified that the tables did have data and worked on their own (they do). Then switched the mast table to compendia instead of local tables and deleted the local tables thinking it might be confusing the two. I tested the master table directly and it also gave the same error some of the time. Every now and again it would give an item and it was from each of the two tables so no obvious reason there I can see. It is giving no error in the console at that level. Tried again several times on the loot sheet to see if anything would come through and it never did and does produce an error in the console. Is this possibly a bug with core foundry?

`lootsheetnpc5e | {this.name} | data {action: 'sheetUpdate', triggerActorId: null, tokenUuid: 'Actor.em8DhkwAwnpaS3tu', processorId: 'LnfrgJkRfBMtTMBR', targetItemId: null, …}
backend.mjs:306 Foundry VTT | Deleted 4 Item documents with ids: [yrzHZsHGlbD0K6qm,DX3E11cxRjXhwrXr,nKh9AdWY9xeeSN6L,jr95DgSz6BK3EqV1] in parent Actor [em8DhkwAwnpaS3tu]
PermissionHelper.js:157 lootsheetnpc5e | PermissionsHelper | getEligablePlayerActors | 8 players with eligable characters found.
LootSeeder.js:97 {itemQtyFormula: '1d3', itemQtyLimitFormula: '3', shopQtyFormula: '1d20', currencyFormula: '1d8[pp],(20d+2)*10 [gp], (10d+1)*10[sp]'}
CurrencyHelper.js:112 lootsheetnpc5e | currencyHelper : Error: Unresolved StringTerm 20d requested for evaluation
    at StringTerm.evaluate (foundry.js:9180:11)
    at Roll._evaluate (foundry.js:6795:42)
    at Roll.evaluate (foundry.js:6762:25)
    at ParentheticalTerm._evaluate (foundry.js:8798:28)
    at ParentheticalTerm.evaluate (foundry.js:7693:25)
    at Roll._evaluate (foundry.js:6782:20)
    at Roll.evaluate (foundry.js:6762:25)
    at Roll.roll (foundry.js:6862:17)
    at Function.tryRoll (CurrencyHelper.js:110:51)
    at Function.generateCurrency (CurrencyHelper.js:64:53)
    at async LootProcessor.buildResults (LootProcessor.js:41:14)
    at async Function.addLootToTarget (ActorHelper.js:171:17)
    at async Function.seedItemsToActors (LootSeeder.js:35:109)
    at async SheetListener.inventoryUpdateListener (SheetListener.js:164:62)
tryRoll @ CurrencyHelper.js:112
await in tryRoll (async)
generateCurrency @ CurrencyHelper.js:64
await in generateCurrency (async)
buildResults @ LootProcessor.js:41
addLootToTarget @ ActorHelper.js:171
await in addLootToTarget (async)
seedItemsToActors @ LootSeeder.js:36
await in seedItemsToActors (async)
inventoryUpdateListener @ SheetListener.js:167
await in inventoryUpdateListener (async)
(anonymous) @ SheetListener.js:90
CurrencyHelper.js:112 lootsheetnpc5e | currencyHelper : Error: Unresolved StringTerm 10d requested for evaluation
    at StringTerm.evaluate (foundry.js:9180:11)
    at Roll._evaluate (foundry.js:6795:42)
    at Roll.evaluate (foundry.js:6762:25)
    at ParentheticalTerm._evaluate (foundry.js:8798:28)
    at ParentheticalTerm.evaluate (foundry.js:7693:25)
    at Roll._evaluate (foundry.js:6782:20)
    at Roll.evaluate (foundry.js:6762:25)
    at Roll.roll (foundry.js:6862:17)
    at Function.tryRoll (CurrencyHelper.js:110:51)
    at Function.generateCurrency (CurrencyHelper.js:64:53)
    at async LootProcessor.buildResults (LootProcessor.js:41:14)
    at async Function.addLootToTarget (ActorHelper.js:171:17)
    at async Function.seedItemsToActors (LootSeeder.js:35:109)
    at async SheetListener.inventoryUpdateListener (SheetListener.js:164:62)
tryRoll @ CurrencyHelper.js:112
await in tryRoll (async)
generateCurrency @ CurrencyHelper.js:64
await in generateCurrency (async)
buildResults @ LootProcessor.js:41
addLootToTarget @ ActorHelper.js:171
await in addLootToTarget (async)
seedItemsToActors @ LootSeeder.js:36
await in seedItemsToActors (async)
inventoryUpdateListener @ SheetListener.js:167
await in inventoryUpdateListener (async)
(anonymous) @ SheetListener.js:90
foundry.js:11928 Foundry VTT | Constructed index of foundry_community_tables.community-tables-5e Compendium containing 1 entries
foundry.js:11928 Foundry VTT | Constructed index of foundry_community_tables.community-tables-independent Compendium containing 184 entries
foundry.js:11928 Foundry VTT | Constructed index of lootsheetnpc5e.lootsheetnpc5e-rolltables Compendium containing 13 entries
foundry.js:11928 Foundry VTT | Constructed index of my-shared-content.shared-roll-tables Compendium containing 1188 entries
foundry.js:11928 Foundry VTT | Constructed index of token-mold.adjectives Compendium containing 7 entries
PermissionHelper.js:157 lootsheetnpc5e | PermissionsHelper | getEligablePlayerActors | 8 players with eligable characters found.`
